src/stream-analytics/azext_stream_analytics/custom.py

```diff
@@ -0,0 +1,206 @@
+    # pylint: disable=no-member
+    return client.resource_groups.get(resource_group_name).location
+
+
+def create_stream_analytics_job(cmd, client, resource_group, name, location=None, tags=None, sku_name='Standard',
+                                output_error_policy=None, events_outoforder_policy=None,
+                                events_outoforder_max_delay=None, events_late_arrival_max_delay=None, data_locale=None,
+                                compatibility_level='1.0', no_wait=False):
+
+    body = {
+        'location': location or _get_resource_group_location(cmd.cli_ctx, resource_group),
+        'tags': tags,
+        'properties': {
+            'sku': {'name': sku_name},
+            'output_error_policy': output_error_policy,
+            'events_out_of_order_policy': events_outoforder_policy,
+            'events_out_of_order_max_delay_in_seconds': events_outoforder_max_delay,
+            'events_late_arrival_max_delay_in_seconds': events_late_arrival_max_delay,
+            'data_locale': data_locale,
+            'compatibility_level': compatibility_level
+        }
+    }
+
+    return sdk_no_wait(no_wait, client.create_or_replace, streaming_job=body, resource_group_name=resource_group,
+                       job_name=name)
+
+
+def update_stream_analytics_job(cmd, client, resource_group, name, tags=None, output_error_policy=None,
+                                events_outoforder_policy=None, events_outoforder_max_delay=None,
+                                events_late_arrival_max_delay=None, data_locale=None):
+    job = client.get(resource_group_name=resource_group, job_name=name)
+    if tags is not None:
+        job.tags = tags
+    if output_error_policy is not None:
+        job.output_error_policy = output_error_policy
+    if events_outoforder_policy is not None:
+        job.events_out_of_order_policy = events_outoforder_policy
+    if events_outoforder_max_delay is not None:
+        job.events_out_of_order_max_delay_in_seconds = events_outoforder_max_delay
+    if events_late_arrival_max_delay is not None:
+        job.events_late_arrival_max_delay_in_seconds = events_late_arrival_max_delay
+    if data_locale is not None:
+        job.data_locale = data_locale
+
+    return client.create_or_replace(streaming_job=job, resource_group_name=resource_group, job_name=name)
+
+
+def delete_stream_analytics_job(cmd, client, resource_group, name, no_wait=False):
+    return sdk_no_wait(no_wait, client.delete, resource_group_name=resource_group, job_name=name)
+
+
+def get_stream_analytics_job(cmd, client, resource_group, name, expand=False):
+    expand = EXPANDED_ALL_STRING if expand else None
+    return client.get(resource_group_name=resource_group, job_name=name, expand=expand)
+
+
+def list_stream_analytics_job(cmd, client, resource_group=None, expand=False):
+    expand = EXPANDED_ALL_STRING if expand else None
+    if resource_group is None:
```

Comment:
Use `if not resource_group_name:` in order to use the workaround `-g=` or `-g ""` to temparorily disable appending the default resource group.

Reply:
updated